database/node.go

```diff
@@ -16,19 +16,26 @@ import (
 	"sort"
 	"strings"
 
+	// Note that this is a deliberate use of the same YAML dependency
+	// as go-dqlite. It preserves files as written by that library,
+	// whereas gopkg.in/yaml.v3 does not.
```

Comment:
Order of fields is not preserved, and the field names become lower-cased.
We basically can't preserve the prior contents when we do a round-trip.

Reply:
I'm not bothered about the order of fields not being preserved, as that's fine. Everything is a map and order doesn't matter. What worries me is that it can potentially lowercase field names, as YAML is case-sensitive.

Reply:
yaml.v3 has an explicit comment about lower-casing the fields.